tests/contenttypes_tests/tests.py

```diff
@@ -378,6 +378,29 @@ class InvalidBookmark(models.Model):
         self.assertEqual(errors, expected)
 
 
+class GenericRelationshipModelTests(TestCase):
```

Comment:
I'm not sure about the organization here. It would be nice to first do a PR that splits up the existing file into things like `test_checks.py`, `test_views.py`, etc.